### PR TITLE
New JS based Robot UI

### PR DESCRIPTION
**Pull request by pmuetschard**
Still very much a work in progress, but shows the main Robot grid view and allows inspecting the trace/report/replay results.

### CODE REVIEW

File: cmd/embed/main.go [hunk 1]
```diff
@@ -120,13 +121,25 @@ func run(ctx context.Context) error {
 		}
 	}
 	var err error
+	usedNames := make(map[string]bool)
```
Comment (ben-clayton) [inline cmd/embed/main.go]: Nit: we more commonly use `map[string]bool{}` if you're not going to specify an initial capacity

File: test/robot/web/build.go [hunk 2]
```diff
@@ -54,6 +57,38 @@ func (s *Server) handlePackages(w http.ResponseWriter, r *http.Request) {
...
+func (s *Server) handlePackageChain(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	head := r.FormValue("head")
+	if head == "" {
+		writeError(w, 404, errors.New("The head parameter is required"))
+		return
+	}
+
+	pkgs := make(map[string]*build.Package)
```
Comment (ben-clayton) [inline test/robot/web/build.go]: nit: `pkgs := map[string]*build.Package{}`

File: test/robot/web/build.go [hunk 3]
```diff
@@ -54,6 +57,38 @@ func (s *Server) handlePackages(w http.ResponseWriter, r *http.Request) {
...
+	if err := s.Build.SearchPackages(ctx, q.Bool(true).Query(), func(ctx context.Context, entry *build.Package) error {
+		pkgs[entry.Id] = entry
+		return nil
+	}); err != nil {
+		writeError(w, 500, err)
+		return
+	}
+
+	pkg := pkgs[head]
```
Comment (ben-clayton) [inline test/robot/web/build.go]: nit: Use `pkg, ok := pkgs[head]` and test `ok`, unless you expect `SearchPackages` to return `nil` for `entry`.

File: test/robot/web/grid.go [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+	"github.com/google/gapid/test/robot/job"
+	"github.com/google/gapid/test/robot/replay"
+	"github.com/google/gapid/test/robot/report"
+	q "github.com/google/gapid/test/robot/search/query"
+	"github.com/google/gapid/test/robot/trace"
+)
+
+type cell struct {
+	Scheduled int
```
Comment (ben-clayton) [inline test/robot/web/grid.go]: Don't use public fields with private structs unless you're doing horrible reflection stuff.

File: test/robot/web/www/js/actions.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
+/*
```
Comment (ben-clayton) [inline test/robot/web/www/js/actions.js]: If you haven't already, please can you run these javascript files through JSLint? There's a [VSCode plugin](https://github.com/ajhyndman/vscode-jslint).

File: test/robot/web/grid.go (hunk 4 above)
Comment (pmuetschard) [reply test/robot/web/grid.go]: This gets serialized to JSON, which does the horrible reflection stuff.

File: test/robot/web/www/js/actions.js (hunk 5 above)
Comment (pmuetschard) [reply test/robot/web/www/js/actions.js]: jslint doesn't like classes

**Pull request merged**